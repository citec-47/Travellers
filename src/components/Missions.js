/* eslint-disable import/extensions */
import React, { useState } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import {
  fetchMissions,
  misionsStatus,
  selectsAllMissions,
} from '../redux/mission/misionSlice';
import style from './mission.module.css';

import Button from '../missionButton/Boton';

const Mission = () => {
  const { mission } = useSelector(selectsAllMissions);
  const dispatch = useDispatch();
  React.useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);
  const [reservation, setIsreservation] = useState(false);

  const handlemissions = (id) => {
    dispatch(misionsStatus(id));
    setIsreservation(!reservation);
  };
  return (
    <div className={style.container}>
      <table>
        <thead>
          <tr>
            <th>Mission</th>
            <th>Description</th>
            <th className={style.stutus}>Status</th>
            <th className={style.but_join}> </th>
          </tr>
        </thead>
        <tbody>
          {mission.map((mission, e) => (
            <tr
              key={mission.mission_id}
              className={e % 2 === 0 ? style.white : style.grey}
            >
              <td className={style.title}>{mission.mission_name}</td>
              <td className={style.desc}>{mission.description}</td>
              <td className={style.stutus_update}>
                <span
                  className={mission.reservation ? style.member : style.nmember}
                >
                  {mission.reservation ? 'Active Member' : 'Not a member'}
                </span>
              </td>
              <td className={style.stutus_update}>
                <Button
                  title={mission.reservation ? 'Leave mission' : 'Join Mission'}
                  className={mission.reservation ? style.btn : style.joind}
                  onReserve={() => handlemissions(mission.mission_id)}
                />
              </td>
            </tr>
          ))}
        </tbody>
      </table>
    </div>
  );
};

export default Mission;
