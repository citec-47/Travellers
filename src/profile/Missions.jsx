/* eslint-disable react/prop-types */
import styles from './Mission.module.css';

function MyMissions({ data }) {
  return (
    <div className={styles.ourMission}>
      <h2>My Missions</h2>
      <ul>
        {data.map((mission) => (
          <li key={mission.id}>{mission.mission_name}</li>
        ))}
      </ul>
    </div>
  );
}

export default MyMissions;
