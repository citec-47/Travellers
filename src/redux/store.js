import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rocket/rocketSlice';
import missionReducer from './mission/misionSlice';

const store = configureStore({
  reducer: {
    allRockets: rocketsReducer,
    mission: missionReducer,
  },
});
export default store;
