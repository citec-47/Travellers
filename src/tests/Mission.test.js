import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Missions from '../components/Missions';
import store from '../redux/store';

describe('Test Missions components', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Missions />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
