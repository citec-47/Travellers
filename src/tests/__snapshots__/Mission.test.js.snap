// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Missions components renders correctly 1`] = `
<div
  className="container"
>
  <table>
    <thead>
      <tr>
        <th>
          Mission
        </th>
        <th>
          Description
        </th>
        <th
          className="stutus"
        >
          Status
        </th>
        <th
          className="but_join"
        >
           
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
